src/lib/features/segment/segment-controller.ts

```diff
@@ -0,0 +1,470 @@
+
+        // Remove unnecessary IFeatureStrategy fields from the response.
+        const segmentStrategies = strategies.map((strategy) => ({
+            id: strategy.id,
+            projectId: strategy.projectId,
+            featureName: strategy.featureName,
+            strategyName: strategy.strategyName,
+            environment: strategy.environment,
+        }));
+
+        res.json({
+            strategies: segmentStrategies,
+        });
+    }
+
+    async removeSegment(
+        req: IAuthRequest<{ id: string }>,
+        res: Response,
+    ): Promise<void> {
+        const id = Number(req.params.id);
+        const strategies = await this.segmentService.getStrategies(id);
+
+        if (strategies.length > 0) {
+            res.status(409).send();
+        } else {
+            await this.segmentService.delete(id, req.user);
+            res.status(204).send();
+        }
+    }
+
+    async updateSegment(
+        req: IAuthRequest<{ id: string }>,
+        res: Response,
+    ): Promise<void> {
+        const id = Number(req.params.id);
+        const updateRequest: UpsertSegmentSchema = {
+            name: req.body.name,
+            description: req.body.description,
+            project: req.body.project,
+            constraints: req.body.constraints,
+        };
+        await this.segmentService.update(id, updateRequest, req.user);
+        res.status(204).send();
+    }
+
+    async getSegment(
+        req: Request<{ id: string }>,
+        res: Response,
+    ): Promise<void> {
+        const id = Number(req.params.id);
+        const segment = await this.segmentService.get(id);
+        if (this.flagResolver.isEnabled('anonymiseEventLog')) {
+            res.json(anonymiseKeys(segment, ['createdBy']));
+        } else {
+            res.json(segment);
+        }
+    }
+
+    async createSegment(
+        req: IAuthRequest<any, any, UpsertSegmentSchema>,
+        res: Response<AdminSegmentSchema>,
+    ): Promise<void> {
+        const createRequest = req.body;
+        const segment = await this.segmentService.create(
+            createRequest,
+            req.user,
+        );
+        this.openApiService.respondWithValidation(
+            201,
+            res,
+            adminSegmentSchema.$id,
+            serializeDates(segment),
+            { location: `segments/${segment.id}` },
+        );
+    }
+
+    async getSegments(
+        req: IAuthRequest,
+        res: Response<SegmentsSchema>,
+    ): Promise<void> {
+        const segments = await this.segmentService.getAll();
+
+        const response = {
+            segments: this.flagResolver.isEnabled('anonymiseEventLog')
+                ? anonymiseKeys(segments, ['createdBy'])
+                : segments,
+        };
+
+        this.openApiService.respondWithValidation<SegmentsSchema>(
+            200,
+            res,
+            segmentsSchema.$id,
+            //@ts-ignore
```

Comment:
I do not understand why typescript complains here. Do you have any input  @thomasheartman?